google/dev/jenkins_task_image_from_deb_repo.sh

```diff
@@ -0,0 +1,91 @@
+#!/bin/bash
+#
+# Copyright 2017 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script creates a Spinnaker GCP image from the Debian packages on Bintray.
+#
+# The following environment variables are expected to be populated:
+#
+# HOME [String] - Home directory for the user running this script.
+#
+# BUILDER_SERVICE_ACCOUNT [String] - Name of the service account used to build and store the image.
+# BUILDER_JSON_CREDENTIALS [String] - Path to the credentials file for $BUILDER_SERVICE_ACCOUNT.
+# BUILD_PROJECT [String] - Name of the GCP project the image is built in.
+# IMAGE_NAME [String] - Name of the resulting Spinnaker image.
+# IMAGE_PROJECT [String] - Name of the GCP project to publish the image to.
+# LOCAL_ZONE [String] - Zone used in this script to build the image. The final resulting image is independent of any zone.
+#
+# DEBIAN_REPO_URL [String] - Public bintray repository to install from. Defaults to the official spinnaker repository.
+# BINTRAY_KEY [String] - Key to use in the case where $DEBIAN_REPO_URL points to a private repository.
+# BINTRAY_USER [String] - Username to authenticate with in the case where $DEBIAN_REPO_URL points to a private repository.
+#
+# GITHUB_REPOSIROTY_BRANCH [String] - Branch within the repository this script should use.
+# GITHUB_REPOSITORY_OWNER [String] - Owner of the 'spinnaker' repository that contains the build_google_image.sh script used within this script.
+
+# Note that this script is GCP-specific -- it only produces GCP Spinnaker images currently.
+#
+# A minimal Jenkins job configuration to run this script:
+#
+# rm -rf spinnaker/
+# git clone https://github.com/spinnaker/spinnaker.git
+# ./spinnaker/dev/jenkins_task_image_from_deb_repo.sh
+
+
+function clone_spinnaker_with_owner() {
+  rm -rf spinnaker
+  if [[ "$GITHUB_REPOSITORY_OWNER" == "default" ]] || [[ "$GITHUB_REPOSITORY_OWNER" == "upstream" ]]; then
+    GITHUB_REPOSITORY_OWNER="spinnaker"
+  fi
+  git clone https://github.com/$GITHUB_REPOSITORY_OWNER/spinnaker.git -b $GITHUB_REPOSITORY_BRANCH
+}
+
+
+function build_spinnaker_google_image() {
+  # Normally these BINTRAY variables arent needed or used.
+  # However, this can give us access to passing a private bintray DEBIAN_REPO_URL
+  export BINTRAY_KEY
+  export BINTRAY_USER
+
+  gcloud config set account $BUILDER_SERVICE_ACCOUNT
```

Comment:
I thought we took out dependencies on the gcloud service account. I see build_google_image.sh uses gcloud to list the current images in order to tease out the base image. That could be a parameter to the script instead and you can compute that here (maybe leaving the default in the script as it is if the parameter isnt set, which is convienent to use by hand as a developer where gcloud account credentials could be assumed). The benefit is that the base image becomes a parameter (i.e. can easly switch to ubuntu 16 or something else entirely as long as it supports debians).

The desire is that running this script should not have a side effect of changing the gcloud account that is configured.